From 5e2b278c5c1eab3e13453ee43403dc03a43fedad Mon Sep 17 00:00:00 2001
From: uweigand <uweigand@138bc75d-0d04-0410-961f-82ee72b054a4>
Date: Mon, 28 Jul 2014 14:33:20 +0000
Subject: [PATCH 144/326] 	PR libobjc/61920 	* encoding.c
 (rs6000_special_adjust_field_align_p): Use definition 	that matches the 4.8
 branch ABI.

git-svn-id: svn+ssh://gcc.gnu.org/svn/gcc/branches/gcc-4_8-branch@213128 138bc75d-0d04-0410-961f-82ee72b054a4
---
 libobjc/ChangeLog  | 6 ++++++
 libobjc/encoding.c | 3 ++-
 2 files changed, 8 insertions(+), 1 deletion(-)

diff --git a/libobjc/ChangeLog b/libobjc/ChangeLog
index b664532..7916009 100644
--- a/libobjc/ChangeLog
+++ b/libobjc/ChangeLog
@@ -1,3 +1,9 @@
+2014-07-27  Ulrich Weigand  <uweigand@de.ibm.com>
+
+	PR libobjc/61920
+	* encoding.c (rs6000_special_adjust_field_align_p): Use definition
+	that matches the 4.8 branch ABI.
+
 2014-07-27  Alan Modra  <amodra@gmail.com>
 	    Matthias Klose  <doko@ubuntu.com>
 
diff --git a/libobjc/encoding.c b/libobjc/encoding.c
index 42b0bf7..a6d1ca6 100644
--- a/libobjc/encoding.c
+++ b/libobjc/encoding.c
@@ -192,7 +192,8 @@ _darwin_rs6000_special_round_type_align (const char *struc, int comp, int spec)
    ? MAX (MAX (COMPUTED, SPECIFIED), 64)				\
    : MAX (COMPUTED, SPECIFIED));})
 
-#define rs6000_special_adjust_field_align_p(FIELD, COMPUTED) 0
+#define rs6000_special_adjust_field_align_p(FIELD, COMPUTED) \
+ (TARGET_ALTIVEC && TREE_CODE (TREE_TYPE (FIELD)) == VECTOR_TYPE)
 
 /* Skip a variable name, enclosed in quotes (").  */
 static inline
-- 
2.0.3

